packages/flutter/lib/src/services/deferred_component.dart

```diff
@@ -0,0 +1,185 @@
+  /// ...
+  /// SplitModule.loadLibrary();
+  /// ```
+  ///
+  /// This method will not load associated dart libraries contained in the dynamic
+  /// feature module, though it will download the files necessary and subsequent
+  /// calls to `loadLibrary()` to load will complete faster.
+  ///
+  /// See also:
+  ///
+  ///  * [uninstallDeferredComponent], a method to request the uninstall of a component.
+  ///  * [loadLibrary](https://api.dart.dev/dart-mirrors/LibraryDependencyMirror/loadLibrary.html),
+  ///    the dart method to trigger the installation of the corresponding deferred component that
+  ///    contains the dart library.
+  ///  * [getDeferredComponentInstallState], a getter that provides the current state
+  ///    of a deferred component installation.
+  static Future<void> installDeferredComponent({required String moduleName}) async {
+    await SystemChannels.deferredComponent.invokeMethod<void>(
+      'installDeferredComponent',
+      <String, dynamic>{ 'loadingUnitId': -1, 'moduleName': moduleName },
+    );
+  }
+  /// Gets the current installation state of the deferred component identified by the
+  /// [moduleName].
+  ///
+  /// This method returns a string that represents the state. Depending on
+  /// the implementation, this string may vary, but the default Google Play
+  /// Store implementation retuns a state in the [DeferredComponentInstallState] enum.
+  ///
+  /// Installations of deferred component modules may be triggered by either calling
+  /// [installDeferredComponent] for assets-only modules or `loadLibrary()` on a deferred
+  /// imported library. Modules not yet requested or do not exist will complete with
+  /// [DeferredComponentInstallState.unknown]. Modules previously installed but not
+  /// loaded in this session will return [DeferredComponentInstallState.installedPendingLoad],
+  /// which indicates that the installation request call should be repeated to
+  /// complete the loading process.
+  ///
+  /// Due to the async nature of platform channels and network i/o, newly requested
+  /// installs may return [DeferredComponentInstallState.unknown] until the installation
+  /// request has been processed. Thus, this state information should be used as purely
+  /// informational and not as an accurate reflection of the readiness of the deferred
+  /// component. Code and assets may only be used after the Future returned by `loadLibrary()`
+  /// or [installDeferredComponent] completes succesfully, regardless of what the
+  /// installation state is in.
+  ///
+  /// See also:
+  ///
+  ///  * [loadLibrary](https://api.dart.dev/dart-mirrors/LibraryDependencyMirror/loadLibrary.html),
+  ///    the dart method to trigger the installation of the corresponding deferred component that
+  ///    contains the dart library.
+  ///  * [installDeferredComponent], a method to install asset-only components.
+  static Future<String?> getDeferredComponentInstallState({required String moduleName}) {
```

Comment:
Can this ever return null? or should the `?` be removed?

Reply:
Also, as said above, it is confusing that this returns a String, but the doc above talks about it returning an enum value of DeferredComponentInstallState.

Reply:
I'm going to just remove this feature from the initial release of deferred components to avoid releasing a strange/limiting API.

The nullable String is because invokeMethod always completes with a nullable value. The value can be null when the method does not exist.